operations/iuf/stages/management_nodes_rollout.md

```diff
@@ -26,22 +35,184 @@ The following arguments are most often used with the `management-nodes-rollout`
+
+    ```bash
+    /usr/share/doc/csm/upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh ncn-w001 --image-id <final_image_id> --desired-cfs-conf <configuration>
+    ```
+
+## Action needed if a worker rebuild fails
+
+In general, worker node rebuilds should complete successfully before starting another rebuild.
+A worker node can get into a bad state if it has been partially rebuilt and an attempt is made to restart the rebuild on that same node.
+As a result, it is not possible to start another worker node rebuild if there is an existing incomplete worker node rebuild workflow, where "incomplete" means it has stopped before successfully completing the full workflow.
+If an incomplete workflow exists and an attempt is made to start another worker rebuild workflow,
+the original incomplete worker rebuild workflow will continue and no new workflow will be created.
 
-When NCN management nodes are being rebuilt, an additional Argo workflow will execute in addition to the standard IUF Argo workflows. It will include the string `ncn-lifecycle-rebuild` in its name.
+If it is necessary to start an entirely new worker rebuild workflow after a previous worker rebuild workflow failed, the failed workflow must be deleted from Kubernetes first.
 
-## Example
+**`WARNING`** Deleting a workflow will delete information about the state of that workflow and the steps that have been completed.
+Deleting a partially complete workflow should be done cautiously and only if needed.
+
+To delete a failed Argo workflow, complete the following steps.
+
+1. Get the name of the failed workflow. All worker rebuild workflows start with `ncn-lifecycle-rebuild`. The name of the worker rebuild workflow can be found in the Argo UI or by searching workflows in Kubernetes with the following command.
+
+    (`ncn-m#`) List failed worker rebuild workflows. Note that NCN storage upgrade workflows will also contain `ncn-lifecycle-rebuild` in their name and may be present in this list.
+
+    ```bash
+    kubectl get workflows -n argo | grep 'ncn-lifecycle-rebuild' | grep 'Fail'
+    ```
+
+1. (`ncn-m#`) Delete the failed workflow.
+
+    ```bash
+    kubectl delete workflows -n argo <failed workflow>
+    ```
+
+    After deleting the failed workflow, a new worker rebuild workflow can be started.
+
+## Examples
 
 (`ncn-m001#`) Execute the `management-nodes-rollout` stage for activity `admin-230127` using the default concurrent management rollout percentage and limiting the operation to `Management_Worker` nodes.
 
 ```bash
 iuf -a admin-230127 run --limit-management-rollout Management_Worker -r management-nodes-rollout
 ```
+
+Expected behavior: All NCN worker nodes will be rebuilt. Each set of worker nodes that is being rebuilt will contain 20% of the total number of worker nodes. For example, if there are 10 total worker nodes, then 2 will be rebuilt at a time.
```

Comment:
Do we have an upper limit for `-cmrp`?    Clearly if that were set to 100 it wouldn't be safe

Reply:
@atifsyedali @JohnNavitskyHPE any idea here?  Pretty sure the CLI doesn't cap it.

Reply:
There is no cap when the percentage is supplied.

Reply:
If there is concern here, we can talk about code changes and then corresponding doc changes once those are in.  For now, I will leave the docs as is for this one.